src/components/EthAddress.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import FontIcon from 'material-ui/FontIcon';
+
+const ETHAddress = ({ ethAddress, networkId }) => {
+  if (networkId === null) {
+    return <div>{ethAddress}</div>;
+  }
+
+  let ethScanPrefix;
+  switch (networkId) {
```

Comment:
This is too much logic, it should be factored out of the view. The view should take an `addressLookupUrl` function that either returns an url or `null`. This function should be provided using a `getAddressLookupUrl` selector in `src/reducers/web3.js`, possibly using `src/chains.json` as a source.